core/thread/inc/ROOT/TReentrantRWLock.hxx

```diff
@@ -123,6 +124,67 @@ struct RecurseCounts {
    size_t GetLocalReadersCount(local_t &local) { return fReadersCount[local]; }
 
 
+};
+
+struct RecurseCountsLocalSpin {
+   using ReaderColl_t = std::unordered_map<std::thread::id, int>;
+   size_t fWriteRecurse; ///<! Number of re-entry in the lock by the same thread.
+
+   std::thread::id fWriterThread; ///<! Holder of the write lock
+   ReaderColl_t fReadersCount;    ///<! Set of reader thread ids
```

Comment:
Why do we need this? `std::thread::id` is probably at least in part responsible for slower performance.
All we should care about are how many readers there are, not which threads are reading.